clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -349,28 +350,23 @@ protected void onJoinComplete(int generation,
         Set<TopicPartition> addedPartitions = new HashSet<>(assignedPartitions);
         addedPartitions.removeAll(ownedPartitions);
 
-        // Invoke user's revocation callback before changing assignment or updating state
-        if (protocol == RebalanceProtocol.COOPERATIVE) {
```

Comment:
the "partitions revoked" should always been empty in the eager case, so it's ok to just check `if (!revokedPartitions.isEmpty()) `